packages/google_maps_flutter/google_maps_flutter_platform_interface/lib/src/types/location.dart

```diff
@@ -59,6 +59,10 @@ class LatLng {
 
   @override
   int get hashCode => Object.hash(latitude, longitude);
+
+  /// Create a [WeightedLatLng] from this [LatLng] with the specified [weight].
+  WeightedLatLng weighted([double weight = 1.0]) =>
```

Comment:
Why does this method exist? Generally classes shouldn't know about classes that compose them.

Reply:
It's just for convenience

Reply:
We should remove it then; inverting layering and adding API complexity is too high a cost for (minimal) convenience. `WeightedLatLng(foo, 0.5)` isn't really less convenient than `foo.weighted(0.5)`.